### PR TITLE
Fix valgrind timing issue failure in replica-redirect test

### PR DESCRIPTION
**Pull request by gmbnomis**
Wait for the replica to become online before starting the actual test.

### CODE REVIEW

File: tests/integration/replica-redirect.tcl [hunk 1]
```diff
@@ -39,6 +39,10 @@ start_server {tags {needs:repl external:skip}} {
         } {}
 
         test {client paused during failover-in-progress} {
+            # Ensure that replica is online and in sync
+            r -1 set foo baz
+            r -1 wait 1 0
```
Comment (enjoy-binbin) [inline tests/integration/replica-redirect.tcl]: we usually use `wait_for_sync` or `wait_for_ofs_sync` to do this.

Comment (gmbnomis) [reply tests/integration/replica-redirect.tcl]: Interestingly, both of these don't do the trick in this case (when running the Valgrind check). We do the equivalent of `wait_for_sync` a couple of lines before and adding a `wait_for_sync`  or `wait_for_ofs_sync` here does not help.

The following `FAILOVER` command that fails during the Valgrind check specifically looks for: 

```c
        if (replica->repl_state != REPLICA_STATE_ONLINE) {
```

But at this point in time, we have:

```
slave0:ip=127.0.0.1,port=21112,state=wait_bgsave,offset=0,lag=0,type=replica
```

Comment (enjoy-binbin) [reply tests/integration/replica-redirect.tcl]: ok, so how about wait_replica_online?

Comment (gmbnomis) [reply tests/integration/replica-redirect.tcl]: Perfect, thanks!

I removed the equivalent of `wait_for_sync` above now and replaced it with `wait_replica_online`

**Review by enjoy-binbin — APPROVED** [review]

**Pull request merged**